tools/js2c.cc

```diff
@@ -381,18 +360,48 @@ std::string GetVariableName(const std::string& id) {
   return result;
 }
 
-std::vector<std::string> GetCodeTable() {
-  size_t size = 1 << 16;
-  std::vector<std::string> code_table(size);
-  for (size_t i = 0; i < size; ++i) {
-    code_table[i] = std::to_string(i) + ',';
+// 382106 is the length of the string "0,1,2,3,...,65535,".
+// 65537 is 2**16 + 1
+constexpr std::pair<std::array<char, 382106>, std::array<uint32_t, 65537>>
```

Comment:
I am not sure if this fares well in MSVC - have you checked how long it compiles + runs on Windows? JS2C is a tool that only gets used at build time. So if we make its build time, say, 100ms slower, so that it runs 10ms faster, then all we get is only the build would be 90ms slower.

Reply:
PS: This also uses 3x less memory.

Reply:
Runtime memory, or compile time memory? The runtime memory should be negligible already, while MSVC might have a hard time compiling this (sometimes MSVC gets killed during compilation due to OOM).

Reply:
I am skeptical whether reducing ~66MB memory usage to ~22MB does anything to the build stability - the memory used to compile/link this file could be bigger than 66MB already. But as long as it doesn't do too much at compile time to risk the compiler running OOM, it doesn't seem to hurt either.